cmd/routed-eni-cni-plugin/cni.go

```diff
@@ -232,6 +232,9 @@ func add(args *skel.CmdArgs, cniTypes typeswrapper.CNITYPES, grpcClient grpcwrap
 
 	result := &current.Result{
 		IPs: ips,
+		Interfaces: []*current.Interface{
+			&current.Interface{Name: hostVethName},
+		},
```

Comment:
Looking at the [CNI SPEC](https://github.com/containernetworking/cni/blob/master/SPEC.md#ips), it states that if an `interfaces` list is returned, the `ip` should have an `interface` index to match. In this case, it would mean adding `Interface: 0,` after line 229, since we only return one interface name.

Reply:
I've mostly been doing manually testing, but this is the config I'm running with at the moment. (which I copied from the docs https://kubernetes.io/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins/#support-traffic-shaping)

```JSON
{
  "cniVersion": "0.3.1",
  "name": "aws-cni",
  "plugins": [
    {
      "name": "aws-cni",
      "type": "aws-cni",
      "vethPrefix": "eni"
    },
    {
      "type": "portmap",
      "capabilities": {"portMappings": true},
      "snat": true
    },
    {
      "type": "bandwidth",
      "capabilities": {
        "bandwidth": true
      }
    }
  ]
}
```

Also maybe unrelated, but I'm using Amazon Linux 2 and I've had to patch this line out, https://github.com/containernetworking/plugins/blob/master/plugins/meta/bandwidth/ifb_creator.go#L34

Because the MTU it's trying to set here is 9001, and the highest it'll accept is 1500. Otherwise seems to work with these annotations on pods. (maybe you understand more?)

```YAML
kubernetes.io/egress-bandwidth: 100M
kubernetes.io/ingress-bandwidth: 100M
```